### PR TITLE
applied tracing propagation for eventing 1.3.x release version

### PR DESCRIPTION
**Pull request by astelmashenko**
This is the same changes as [PR 342 implemented propagating w3c tracing context headers via stan channels](https://github.com/knative-sandbox/eventing-natss/pull/342)
What is different:
- e2e eventing version
- golangci_lint version

### CODE REVIEW

**Comment by astelmashenko** [comment]
@lionelvillard ,
Intent of this PR is to fix tracing propagation for eventing-natss 1.3.x release. I marked this PR as draft because I'm not sure which branch to merge to, is target branch [knative-sandbox:release-1.3](https://github.com/knative-sandbox/eventing-natss/tree/release-1.3) correct?

**Comment by astelmashenko** [comment]
@lionelvillard ,
There is a codecov error, but honestly I do not know how to test it tracing.SetupDynamicPublishing will always return nil error. Do we need to test that?
If yes please point me how I can do it, may be some documentation on how to test it.

**Comment by astelmashenko** [comment]
@lionelvillard , all checks pass now. Please help with releasing 1.3.1.
I've read https://github.com/knative/release/blob/main/PROCEDURES.md but looks like to main stream release. What else should be done for patch-release?
Thanks

**Comment by astelmashenko** [comment]
I'm going to create a PR from a cherrypick of merged PR. Stay tuned

**Comment by astelmashenko** [comment]
Re-openning, since I can not create cherry-pick PR.

File: pkg/tracing/tracing.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	additionalHeaders.Add(traceStateHeader, ts)
+
+	return additionalHeaders
+}
+
+func ConvertNatssMsgToEvent(logger *zap.Logger, stanMsg *stan.Msg) *event.Event {
+	message := cloudevents.NewEvent()
+	err := json.Unmarshal(stanMsg.Data, &message)
+	if err != nil {
+		logger.Error("could not create a event from stan msg", zap.Error(err))
```
Comment (lionelvillard) [inline pkg/tracing/tracing.go:81]: nit: create **an** event

Comment (astelmashenko) [reply pkg/tracing/tracing.go]: fixed

**Comment by lionelvillard** [comment]
/lgtm

Thanks!!

**Comment by astelmashenko** [comment]
@lionelvillard , is anything else needed to merge this PR?

**Comment by astelmashenko** [comment]
hi, @lionelvillard, when does it get released? I mean 1.3.1?
Thank you

**Comment by lionelvillard** [comment]
Since we removed this repo from the release train due the lack of maintainers, it also disabled the all types of releases. Since you offered (thanks!) to be a maintainer, I'm gonna add it again to the release train.

**Pull request merged**